### PR TITLE
Enable --dart-flags in Android engine

### PR DESCRIPTION
**Pull request by johnmccutchan**

### CODE REVIEW

**Comment by johnmccutchan** [comment]
Pairs with: https://github.com/flutter/flutter/pull/2680

**Comment by johnmccutchan** [comment]
@abarth

File: sky/shell/platform/android/org/domokit/sky/shell/SkyActivity.java [hunk 1]
```diff
@@ -48,6 +48,10 @@
         if (intent.getBooleanExtra("start-paused", false)) {
             args.add("--start-paused");
         }
+        String dartFlags = intent.getStringExtra("dart-flags");
+        if (dartFlags != null) {
+          args.add("--dart-flags=\"" + dartFlags + "\"");
+        }
```
Comment (abarth) [inline sky/shell/platform/android/org/domokit/sky/shell/SkyActivity.java]: Please read the comment on line 38.

**Comment by abarth** [comment]
Unfortunately, this code will be a security vulnerability in production.  We need some way of disabling this code in production or we need to find some other way to address this use case.

**Comment by johnmccutchan** [comment]
@mhausner FYI

**Comment by abarth** [comment]
We've talked about being more specific about "Debug", "Profiling", and "Release" builds.  Maybe we need to make that real and scope this mechanism to "Debug" and "Profiling" builds.

**Comment by johnmccutchan** [comment]
@abarth That sounds reasonable to me

**Comment by eseidelGoogle** [comment]
https://github.com/flutter/flutter/issues/2042 is at least one issue relating to the debug/deploy/profiling separation.  I believe there is a metabug, but I can't find it at the moment.

**Comment by devoncarew** [comment]
https://github.com/flutter/flutter/issues/2366

**Comment by Hixie** [comment]
I discussed this with @johnmccutchan and he said this should be closed for now and he'd get back to it.

**Pull request closed without merging**